turbinia/workers/containerd.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for analyzing containerd containers."""
+
+from __future__ import unicode_literals
+
+import json
+import logging
+import subprocess
+import tempfile
+
+from turbinia import config
+from turbinia import TurbiniaException
+from turbinia.evidence import ContainerdContainer
+from turbinia.evidence import EvidenceState as state
+from turbinia.workers import Priority
+from turbinia.workers import TurbiniaTask
+
+log = logging.getLogger('turbinia')
+
+CE_BINARY = '/opt/container-explorer/bin/ce'
+CE_SUPPORT_FILE = '/opt/container-explorer/etc/supportcontainer.yaml'
+
+
+class ContainerdEnumerationTask(TurbiniaTask):
+  """Enumerate containerd containers on Linux."""
+
+  REQUIRED_STATES = [state.ATTACHED, state.MOUNTED]
+
+  def list_containers(self, evidence, _, detailed_output=False):
+    """List containerd containers in the evidence.
+
+    Args:
+      evidence (Evidence): Input evidence to be processed.
+      result (TurbiniaTaskResult): Object to store logs.
+
+    Returns:
+      list(dict): Containers information
+    """
+    config.LoadConfig()
+    containers = None
+    image_path = evidence.mount_path
+
+    outputfile = tempfile.mkstemp()[1]
+    list_cmd = [
+        'sudo', CE_BINARY, '--support-container-data', CE_SUPPORT_FILE,
+        '--output', 'json', '--output-file', outputfile, '--image-root',
+        image_path, 'list', 'containers'
+    ]
+    log.info(f'Running {list_cmd}')
+
+    try:
+      #json_data = subprocess.check_output(list_cmd).decode('utf-8')
+      subprocess.check_call(list_cmd)
+      with open(outputfile, 'r') as fh:
+        json_data = fh.read()
+        if json_data:
+          containers = json.loads(json_data)
+    except json.JSONDecodeError as e:
+      raise TurbiniaException(
+          f'Error decoding container-explorer output: {e}') from e
+    except subprocess.CalledProcessError as e:
```

Comment:
Might want to add `except FileNotFoundError` as well in case the `outputfile` doesn't exist for some reason.

Reply:
Exception added to handle missing outputfile.